### DISCUSSION TITLE
Interactive Console with StepObject

### DISCUSSION THREAD

**Discussion by cjhille** [Codeception/Codeception#6174] [Ideas]
Is there an easy way to get the interactive console to include methods defined in a StepObeject? This would be quite useful in case the StepObject Class contains common functions like login or session related code. Using the acceptance helper class instead might work, but I'm not sure whether that's a good practice.

**Comment by DavertMik** [comment]
Currently there is no way to do so. But thanks for the idea, I will try to implement this soon

**Comment by cjhille** [comment]
Thanks Michael. I'm excited to hear that. 
Just to follow up on the enhanced support for objects within the interactive console, I think it would also be really useful to improve support for Page Objects as well.
In larger test suites it makes sense to use Page Objects not only for storage of static locators within a page, but also as an interface for user actions with that page ([Fowler 2013](http://martinfowler.com/bliki/PageObject.html)).
For instance, if I wanted to use a convenience method to fill out an order form from console, it could be accessed by using `OrderPage::of($I)->fillOutForm()`, which unfortunately is not supported by the console, although the Page Objects are generally accessible from console. Maybe this could be done be checking if the $command is preceded by a static method, or by making the actor instance `$I` removable from input (maybe by inserting it as user input in command dialog).
